drivers/pinctrl/pinctrl_emsdp.c

```diff
@@ -261,6 +267,7 @@ static int pinctrl_emsdp_set(uint32_t pin, uint32_t type)
 			reg |= ARDUINO_CFG6_I2C;
 		}
 		break;
+	case INNER_CONNECT:
```

Comment:
Let's avoid reading / writing MMIO when not required - we can just add

```c
if (pin == INNER_CONNECT) {
	return 0;
}
```

to the beggining of pinctrl_emsdp_set.